### PR TITLE
diffutils:  fix bug with +universal variant

### PR DESCRIPTION
**Pull request by RobK88**
#### Description

* Bump revision number
* Add pre-configure{} code block to add the "--disable-year2038" configure flag for +universal builds
* Change "configure.args" to "configure.args-append"
* Add "--disable-silent-rules" flag to "configure.args-append"
* Reformat Portfile

CLOSES:  https://trac.macports.org/ticket/67475

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G14042 x86_64
Xcode 10.1 10B61

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by ryandesign** [comment]
Please don't reformat the portfile and make functional changes all within one commit. It's impossible to see what functional changes you're proposing when you do that. If you must reformat, do so in a separate commit.

Don't use `--disable-year2038` only in the universal variant. It is needed for pure 32-bit builds as well. It should be harmless to use this flag for all builds. All this flag does is prevent the configure script from complaining when it detects a 32-bit build is being performed.

There is no need to increase the revision since it does not change how the port installs, in those situations where it could install before.

**Comment by ryandesign** [comment]
> CLOSES: https://trac.macports.org/ticket/67475

Please add `Closes: https://trac.macports.org/ticket/67475` to the commit message (not just the PR description) so that Trac will close the ticket when this PR is merged.

**Review by mascguy — APPROVED** [review]
LGTM. I'll update the commit message on merge, per Ryan's previous comment.

**Comment by RobK88** [comment]
@masguy @ryandesign - Thanks for the feedback.. I will remember to add "CLOSES: PR#" to my commit messages in the future and will try to avoid reformatting other people's Portfiles.

File: sysutils/diffutils/Portfile [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
 
-PortGroup           compiler_blacklist_versions 1.0
+PortGroup                   compiler_blacklist_versions 1.0
 
-name                diffutils
-version             3.10
-revision            0
+name                        diffutils
+version                     3.10
+revision                    1
```
Comment (jmroot) [inline sysutils/diffutils/Portfile:9]: The revision shouldn't be increased when fixing a problem that prevented installation.

File: sysutils/diffutils/Portfile [hunk 2]
```diff
@@ -51,20 +50,26 @@ use_xz              yes
...
+depends_lib-append          port:gettext-runtime
 
-patchfiles          patch-src_system.h.diff
-configure.args      --program-prefix=g \
-                    --infodir=${prefix}/share/info \
-                    --mandir=${prefix}/share/man
+patchfiles                  patch-src_system.h.diff
 
+# Bypass year 2038 awareness code that causes the universal variant to fail.
+pre-configure {
```
Comment (jmroot) [inline sysutils/diffutils/Portfile:62]: There's no need to do this in pre-configure. As Ryan pointed out, the flag can be used unconditionally, but even if it should be used only when building universal, the appropriate check is:
```
if {$universal_possible && [variant_isset universal]} {
…
}
```

File: sysutils/diffutils/Portfile [hunk 3]
```diff
@@ -51,20 +50,26 @@ use_xz              yes
...
+                            if [variant_isset "universal"] {
```
Comment (jmroot) [inline sysutils/diffutils/Portfile:63]: Remember to [brace your expressions](https://wiki.tcl-lang.org/page/Brace+your+expr%2Dessions).

File: sysutils/diffutils/Portfile (hunk 2 above)
Comment (RobK88) [reply sysutils/diffutils/Portfile:62]: Thanks for the feedback.  I was not sure how this additional new flag would affect 64-bit builds.  That was the main reason I attempted to limit it to universal builds.  And thank you for the code.  I will remember it next time I ever need to limit my code to universal builds.

File: sysutils/diffutils/Portfile (hunk 3 above)
Comment (RobK88) [reply sysutils/diffutils/Portfile:63]: Thanks.  I am still learning tcl.

**Comment by RobK88** [comment]
FYI -- I submitted another PR to fix 32-builds as noted by @ryandesign 

See
https://github.com/macports/macports-ports/pull/18890

**Pull request merged**